src/main/java/com/spikes2212/control/RioPIDLoop.java

```diff
@@ -147,7 +147,7 @@ public void enable() {
 
     private void periodic() {
         synchronized (controller) {
```

Comment:
please replace this `syncronized` block with a use of a `ReetrantLock` you can read about it [here](https://www.geeksforgeeks.org/reentrant-lock-java/).
you will need to store a private instance of `ReentrantLock`.